lib/providers/user.dart

```diff
@@ -1,9 +1,46 @@
+import 'dart:async';
+import 'dart:convert';
 import 'package:flutter/material.dart';
+import '../util.dart';
 
 class User with ChangeNotifier {
   String _idToken; // Firebase ID token of the account.
   String _localId; // The uid of the current user.
   String email;
   String displayName;
   String photoUrl;
+
+  Future<Map<String, dynamic>> getDeviceIP() async {
+    try {
+      var responseDeviceIP = await Util.fetch(FetchType.GET, 'http://ip-api.com/json');
+      final responseData = json.decode(responseDeviceIP) as Map<String, dynamic>;
+      return responseData;
+    } catch (error) {
+      return {'error': error};
+    }
+  }
+
+  Future<Map<String, dynamic>> getOfficeIP() async {
```

Comment:
we've already had fetchOfficeSettings() from Auth Providers, pls use ProxyProvider to inject this function from Auth Provider to User Provider